.github/workflows/test-all-packages.yml

```diff
@@ -78,6 +78,43 @@ jobs:
     - name: lint check
       run: yarn lint-check
 
+  benchmark:
+    needs: build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: ['14.x']
+    if: ${{github.event_name == 'push' && github.ref == 'refs/heads/master'}}
+    steps:
+    - uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    # BEGIN-RESTORE-BOILERPLATE
+    - name: restore built files
+      id: built
+      uses: actions/cache@v1
+      with:
+        path: .
+        key: ${{ runner.os }}-${{ matrix.node-version }}-built-${{ github.sha }}
+    - uses: actions/checkout@v1
+      with:
+        submodules: 'true'
+      if: steps.built.outputs.cache-hit != 'true'
```

Comment:
@michaelfig if this step is `needs: build`, would it ever be the case that `cache-hit == false`? I'm slightly worried that having steps that do build-like actions *at all* in this job, when there are ordering rules intended to make sure they're never used, is an invitation for bitrot, as the `build` job evolves, but this unloved unused copies of `yarn install`/etc do not. I'm thinking it might be better to omit the checkout/yarn-install/yarn-build steps entirely, from every job except for `build`.

Reply:
I think the checkout/install/build stuff is to deal with a cache miss (which presumably can happen since it's, you know, a cache).

Reply:
> I'm thinking it might be better to omit the checkout/yarn-install/yarn-build steps entirely, from every job except for `build`

Github Actions jobs are totally independent.  `needs: build` only sets the order in which they are run.  The only way to communicate build results from `build` to other jobs is via a cache of some kind.  The cache can be missed occasionally.  If it didn't happen to me, I would not have implemented this complexity.

I agree that the Actions are not DRY.  If you're really concerned about the boilerplate, complain to Github (who has since last time I checked, been refusing to implement macros even though YAML supports them) or maybe implement a script that can update all the jobs.  This is not the only file with repetition.